plugins/device-injector/device-injector.go

```diff
@@ -162,6 +190,36 @@ func parseDevices(ctr string, annotations map[string]string) ([]device, error) {
 	return devices, nil
 }
 
+func parseCDIDevices(ctr string, annotations map[string]string) ([]string, error) {
+	var (
+		key        string
+		annotation []byte
+		cdiDevices []string
+	)
+
+	// look up effective device annotation and unmarshal devices
```

Comment:
Refactor idea: Getting the list of annotations is common to devices, mounts, and CDI devices and should be factored out into a separate function.

Reply:
Implemented in the refactored version.